Codes update 04/23

hzyguan888 opened the pull request:
Please help merge this branch

wengxyu1030 committed:
Merge pull request #1 from wengxyu1030/hzyguan888-patch-1

c_hospdel

hzyguan888 committed:
Codes update, 4/23

I updated some codes, and may need your review.

Also, I had some questions for 16_woman_cancer. Looking forward to your reply!

wengxyu1030 commented on 16_woman_cancer.do:
```diff
@@ -11,8 +11,11 @@ if _rc==0 {
     replace w_mammogram=. if s1017==. | s1017==9 | s1020==9
 }
+
+// I'm wodering where do you find these codes? 
```
It's from the recode VI adeptfile. So far I didn't see any different code on this one in the country specific case.

wengxyu1030 replied:
It's from the recode VI adeptfile. So far I didn't see any different code on this one in the country specific case.

wengxyu1030 commented on 16_woman_cancer.do:
```diff
@@ -11,8 +11,11 @@ if _rc==0 {
+
+// I'm wodering where do you find these codes? 
+// Be careful these two variables may be country specific in recode six too.
+// Also may need to add them in the quality control file, as they could be compared with HEFPI database.
```
Please do and update in the Github for this indicator in the Quality control code.

wengxyu1030 commented on 1_antenatal_care.do:
```diff
@@ -6,17 +6,7 @@
 rename *,lower
 order *,sequential
 
-    *anc_skill: Categories as skilled: doctor, nurse, midwife, auxiliary nurse/midwife...
```
Can you explain what is the change you make?

wengxyu1030 commented on 1_antenatal_care.do:
```diff
@@ -61,7 +65,7 @@ order *,sequential
 	
 	*c_anc_bp: Blood pressure measured during pregnancy of births in last 2 years
-	clonevar c_anc_bp = m42c 
+	clonevar c_anc_bp = m42c            // please check this as well as bs and ur with my codes
```
Are you saying this might be country specific?

hzyguan888 replied:
I found a filter question in recode seven. It may be the same for recode six. Please check. And if so, you may need to update the code.

hzyguan888 replied:
Do you change anything on the codes?

wengxyu1030 commented on 3_postnatal_care.do:
```diff
@@ -12,7 +12,7 @@
 	replace  `var'_skill= 1 if ///
 	(regexm(`var'_lab,"doctor|nurse|midwife|aide soignante|assistante accoucheuse|clinical officer|mch aide|trained|auxiliary birth attendant|physician assistant|professional|ferdsher|skilled|community health care provider|birth attendant|hospital/health center worker|hew|auxiliary|icds|feldsher|mch|vhw|village health team|health personnel|gynecolog(ist|y)|obstetrician|internist|pediatrician|family  ...
-	|!regexm(`var'_lab,"na^|-na|traditional birth attendant|untrained|unquallified|empirical midwife")) 
+	&!regexm(`var'_lab,"na^|-na|traditional birth attendant|untrained|unquallified|empirical midwife|other")) 
```
Thanks, great catch!

wengxyu1030 commented on 8_child_illness.do:
```diff
@@ -43,7 +43,7 @@ order *,sequential  //make sure variables are in order.
 
         gen c_diarrhea_pro = 0 if c_diarrhea == 1
-        replace c_diarrhea_pro = 1 if pro >= 1 
+        replace c_diarrhea_pro = 1 if c_diarrhea_pro == 0 & pro >= 1 
```
I think I should rewrite pro as pro_dia here?

hzyguan888 replied:
Oh yes, please do

wengxyu1030 commented on 8_child_illness.do:
```diff
@@ -135,7 +135,7 @@ order *,sequential  //make sure variables are in order.
 		
 		foreach var of varlist c_treatARI c_treatARI2 {
-        replace `var' = 1 if pro_ari >= 1 
+        replace `var' = 1 if `var' == 0 & pro_ari >= 1 
```
Yes, this is more accurate.

wengxyu1030 commented on 8_child_illness.do:
```diff
@@ -202,6 +202,8 @@ order *,sequential  //make sure variables are in order.
 			replace c_fevertreat = 1 if c_fevertreat == 0 & `var' == 1
 			replace c_fevertreat = . if `var' == 9 
 		}
+		// you may need to change $h32 according to the general codes above, as in general codes, we don't have $h32 anymore.
```
Can you write the changes in code? I assume we are still using h32 in the general codes? Why not?

hzyguan888 replied:
Yes, I wrote the changes. do you see them?

wengxyu1030 replied:
Seems you didn't pull request instead of merged to the master. So I could not see where are the changes?

wengxyu1030 committed:
Skilled Worker

This file extracts all the labels for the skilled workers by the survey.

wengxyu1030 committed:
Update skilled_worker.Rmd

wengxyu1030 committed:
Update 16_woman_cancer.do

wengxyu1030 committed:
Update the filter question for c_anc_bp, bs, ur

There's a filter question for the antenatal care for blood, urine sample taken, and pressure measurement.

wengxyu1030 reviewed (approved):
The filter questions are added for the blood sample taken, blood pressure, and urine sample.

wengxyu1030 committed:
Changed pro to pro_dia

That was a typo for pro_dia

hzyguan888 committed:
Update 8_child_illness.do

Please see my changes for $h32 as requested

hzyguan888 committed:
anc_skill=1 if trained traditional birth attendant

Could you assign anc_skill = 1 if it is trained traditional birth attendant? As trained traditional birth attendant should be categorized as skilled.
